x-pack/plugins/infra/server/lib/alerting/inventory_metric_threshold/inventory_metric_threshold_executor.ts

```diff
@@ -115,18 +118,25 @@ export const createInventoryMetricThresholdExecutor = (libs: InfraBackendLibs) =
     }
     if (reason) {
       const actionGroupId =
-        nextState === AlertStates.OK ? RecoveredActionGroup.id : FIRED_ACTIONS.id;
-      alertInstance.scheduleActions(actionGroupId, {
-        group: item,
-        alertState: stateToAlertMessage[nextState],
-        reason,
-        timestamp: moment().toISOString(),
-        value: mapToConditionsLookup(results, (result) =>
-          formatMetric(result[item].metric, result[item].currentValue)
-        ),
-        threshold: mapToConditionsLookup(criteria, (c) => c.threshold),
-        metric: mapToConditionsLookup(criteria, (c) => c.metric),
-      });
+        nextState === AlertStates.OK ? RecoveredActionGroup.id : FIRED_ACTIONS_ID;
+      alertInstance.scheduleActions(
+        /**
+         * TODO: We're lying to the compiler here as explicitly  calling `scheduleActions` on
+         * the RecoveredActionGroup isn't allowed
+         */
+        (actionGroupId as unknown) as InventoryMetricThresholdAllowedActionGroups,
```

Comment:
The `InventoryMetricThreshold` AlertType relies on specifying the recovery ActionGroup explicitly in order to include a _context_ that is accessible when an instance is recovered. This approach isn't in line with how the system was designed to be used, which is why we have encountered some unforeseen side effects (alert instances recovering twice, for example).

This PR makes this kind of usage impossible _unless_ you lie to the compiler, which is why we do so here.

I have began a dialogue with @Zacqary to find a solution that will satisfy their need, but for now, I'd still like to proceed with this PR to discourage this type of usage.

Reply:
@mikecote I couldn't find an existing issue to explore ways of enabling some kind of context when an instance recovers. Are you aware of one which I can link to here?
If not, I'll open a new one and we can try and find a solution for Metrics.

Reply:
@gmmorris none that I recall, just some conversations in another issue: https://github.com/elastic/kibana/issues/49405#issuecomment-722691370.